### PR TITLE
[TRAFODION-1835] Re-factor license files

### PR DESCRIPTION
**Pull request by svarnau**
Enable building license files for source and binary distributions.

Since we build several different distribution packages, the LICENSE
files need to be specific to the components packaged.

The licenses/Makefile indicates how we combine each component source
and binary license files for source package and the various binary
distribution packages.

Binary distribution license info is still incomplete.  This change
sets the framework, and license content will be completed later.
Several of the licenses/*-bin files have "To Do" items indicating
remaining work to be done related to TRAFODION-1816.

### CODE REVIEW

File: licenses/Makefile [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+#
+# @@@ END COPYRIGHT @@@
+
+all: LICENSE-src LICENSE-dcs LICENSE-server LICENSE-install LICENSE-rest LICENSE-clients
+
+# All source code included in Trafodion source
+LICENSE-src:
+	cat Apache lic-components-src lic-dcs-src lic-server-src lic-test-src > $@
+
+# Binary licenses for each separtely packaged component
```
Comment (DaveBirdsall) [inline licenses/Makefile]: Spelling error in comment (fix it later): "separtely"

**Comment by DaveBirdsall** [comment]
Looks good to me.

File: licenses/Makefile (hunk 1 above)
Comment (svarnau) [inline licenses/Makefile]: Fixing now, along with makefile bug.

**Comment by robertamarton** [comment]
Looks good to me but I am not a make expert.

**Comment by svarnau** [comment]
FYI. If reviewers want to look at the resulting tar files, which should include the correct LICENSE info, then you can look at jenkins build artifacts that will be around for a few days: https://jenkins.esgyn.com/job/build-master-release/507/

**Pull request merged**